Maple2.File.Parser/Xml/AI/Condition.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Generic;
+using System.Xml.Serialization;
+
+namespace Maple2.File.Parser.Xml.AI;
+
+public enum ConditionTargetState {
+    grabTarget,
+    holdme
+}
+
+public enum ConditionOp {
+    Equal,
+    Greater,
+    Less,
+    GreaterEqual,
+    LessEqual,
+
+    equal = Equal,
+    greaterEqual = GreaterEqual,
+    lessEqual = LessEqual,
+    greater = Greater,
+    less = Less
+}
+
+public partial class Condition {
+    [XmlAttribute] public string name = string.Empty;
+
+    [XmlElement] public List<Node> node = new List<Node>();
+    [XmlElement] public List<AiPreset> aiPreset = new List<AiPreset>();
+
+    [XmlAttribute] public int value;
+    [XmlAttribute] public int battleTimeBegin;
+    [XmlAttribute] public int battleTimeLoop;
+    [XmlAttribute] public int battleTimeEnd;
+    [XmlAttribute] public int skillIdx;
+    [XmlAttribute] public short skillLev;
```

Comment:
here and elsewhere, I would set default value of "level" to 1 because level 0 never makes sense.

Reply:
One concern I have with this is that both skillIdx & skillId + skillLev live in the same nodes and setting the default to 1 makes it unclear if it is meant to be an override or not. Might be better to set the default to something clearly bad like -1 to indicate if its intentionally set or not. Example:
`<condition name="skillRange" skillIdx="1" skillLev="1">`